### PR TITLE
chore(feat): improve probe bpf helper based on program type

### PR DESCRIPTION
**Pull request by rscampos**
Create type `BPFFunc` in package `libbpfgo` in or to avoid import package `helpers` in libbpfgo.go.
Improve tests in order to use capabilities.

### CODE REVIEW

File: prog-common.go [hunk 1]
```diff
@@ -6,6 +6,9 @@ package libbpfgo
 */
 import "C"
 
+// BPFFunc is an enum as defined in https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/bpf.h
+type BPFFunc uint32
+
```
Comment (geyslan) [inline prog-common.go:11]: I suppose we need to bring here all the consts (enum) together, providing then to the lib consumer.

https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/bpf.h#L5786

https://github.com/aquasecurity/libbpfgo/blob/73f0dea092f125ae0c210623bcaa03449e916086/helpers/argumentParsers.go#L1970

Comment (rscampos) [reply prog-common.go:11]: Good point, got the enum with some helpers functions from the `helpers` package.

**Review by geyslan — APPROVED** [review]
LGTM. Good testing cases! Awesome.

I would only suggest an improved `BPFHelperIsSupported` comment.

File: libbpfgo.go [hunk 2]
```diff
@@ -98,10 +96,18 @@ func BPFMapTypeIsSupported(mapType MapType) (bool, error) {
 	return supportedC == 1, nil
 }
 
-func BPFHelperIsSupported(progType BPFProgType, funcId helpers.BPFFunc) (bool, error) {
-	supportedC := C.libbpf_probe_bpf_helper(C.enum_bpf_prog_type(int(progType)), C.enum_bpf_func_id(int(funcId)), nil)
+// Probe bpf helper func for a given program type
+// Note: BPF returns "Invalid Argument" when helper function is unavailable, just ignore the error.
+func BPFHelperIsSupported(progType BPFProgType, funcId BPFFunc) (bool, error) {
+	supportedC, errno := C.libbpf_probe_bpf_helper(C.enum_bpf_prog_type(int(progType)), C.enum_bpf_func_id(int(funcId)), nil)
```
[the remark this answers is not in the document]
Comment (rscampos) [reply libbpfgo.go:102]: Good catch

**Comment by rscampos** [comment]
Done, thank you for the review @geyslan

**Review by geyslan — APPROVED** [review]

**Pull request merged**